### PR TITLE
[cloud-providers][openstack] Explain the origin of the mainNetwork parameter

### PR DESCRIPTION
**Pull request by hugik-88**
## Description

This PR explains how to generate the mainNetwork parameter for all layouts in the OpenStackClusterConfiguration resource
## Why do we need it, and what problem does it solve?

For many engineers who are new to bootstrapping a cluster on an openstack-based cloud, as well as users, it can be difficult to determine what value to use for the mainNetwork parameter. This can lead to multiple iterations of the cluster bootstrapping process, as options are tested in order to find the correct one. In the PR, we provide an unambiguous definition for this parameter to improve the user experience and ensure a smoother process.
## Why do we need it in the patch release (if we do)?

Not necessarily
## What is the expected result?

The mainNetwork parameter no longer causes difficulties when bootstrapping a cluster to an openstack-based cloud
## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: cloud-provider-openstack
type: chore 
summary: Help the user better understand the bootstrap of the cluster in the OpenStack based cloud.
impact_level:  low
```

### CODE REVIEW

File: ee/candi/cloud-providers/openstack/docs/LAYOUTS.md [hunk 1]
```diff
@@ -74,6 +74,7 @@ nodeGroups:
     # is no dhcp in network that is used as default gateway.
     configDrive: false
     # Required, network will be used as default gateway.
```
Comment (z9r5) [inline ee/candi/cloud-providers/openstack/docs/LAYOUTS.md:76]:
```suggestion
    # Required, the gateway of the network will be used as the default gateway.
```

File: ee/candi/cloud-providers/openstack/docs/LAYOUTS.md [hunk 2]
```diff
@@ -74,6 +74,7 @@ nodeGroups:
...
+    # Matches the cloud.prefix in the ClusterConfiguration resource
```
Comment (z9r5) [inline ee/candi/cloud-providers/openstack/docs/LAYOUTS.md:77]:
```suggestion
    # Matches the cloud.prefix in the ClusterConfiguration resource.
```

File: ee/candi/cloud-providers/openstack/docs/LAYOUTS.md [hunk 3]
```diff
@@ -158,6 +159,7 @@ nodeGroups:
     # gateway.
     configDrive: false
     # Required, network will be used as default gateway.
+    # Matches the cloud.prefix in the ClusterConfiguration resource
```
Comment (z9r5) [inline ee/candi/cloud-providers/openstack/docs/LAYOUTS.md:162]:
```suggestion
    # Required, the gateway of the network will be used as the default gateway.
    # Matches the cloud.prefix in the ClusterConfiguration resource.
```

File: ee/candi/cloud-providers/openstack/docs/LAYOUTS.md [hunk 4]
```diff
@@ -235,6 +237,7 @@ nodeGroups:
...
```
Comment (z9r5) [inline ee/candi/cloud-providers/openstack/docs/LAYOUTS.md:240]:
```suggestion
    # Required, the gateway of the network will be used as the default gateway.
    # Matches the cloud.prefix in the ClusterConfiguration resource.
```

File: ee/candi/cloud-providers/openstack/docs/LAYOUTS.md [hunk 5]
```diff
@@ -315,6 +318,7 @@ nodeGroups:
...
```
Comment (z9r5) [inline ee/candi/cloud-providers/openstack/docs/LAYOUTS.md:321]:
```suggestion
    # Required, the gateway of the network will be used as the default gateway.
    # Matches the cloud.prefix in the ClusterConfiguration resource.
```

File: ee/candi/cloud-providers/openstack/docs/LAYOUTS_RU.md [hunk 6]
```diff
@@ -76,6 +76,7 @@ nodeGroups:
     # качестве шлюза по умолчанию.
     configDrive: false
     # Обязательный параметр, сеть будет использована как шлюз по умолчанию.
+    # Совпадает с сloud.prefix в ресурсе ClusterConfiguration
```
Comment (z9r5) [inline ee/candi/cloud-providers/openstack/docs/LAYOUTS_RU.md:79]:
```suggestion
    # Обязательный параметр, шлюз этой сети будет использован как шлюз по умолчанию.
    # Совпадает с сloud.prefix в ресурсе ClusterConfiguration.
```

File: ee/candi/cloud-providers/openstack/docs/LAYOUTS_RU.md [hunk 7]
```diff
@@ -166,6 +167,7 @@ nodeGroups:
...
```
Comment (z9r5) [inline ee/candi/cloud-providers/openstack/docs/LAYOUTS_RU.md:170]:
```suggestion
    # Обязательный параметр, шлюз этой сети будет использован как шлюз по умолчанию.
    # Совпадает с сloud.prefix в ресурсе ClusterConfiguration.
```

File: ee/candi/cloud-providers/openstack/docs/LAYOUTS_RU.md [hunk 8]
```diff
@@ -248,6 +250,7 @@ nodeGroups:
...
```
Comment (z9r5) [inline ee/candi/cloud-providers/openstack/docs/LAYOUTS_RU.md:253]:
```suggestion
    # Обязательный параметр, шлюз этой сети будет использован как шлюз по умолчанию.
    # Совпадает с сloud.prefix в ресурсе ClusterConfiguration.
```

File: ee/candi/cloud-providers/openstack/docs/LAYOUTS_RU.md [hunk 9]
```diff
@@ -330,6 +333,7 @@ nodeGroups:
     # качестве шлюза по умолчанию.
     configDrive: false
     # Обязательный параметр. Сеть будет использована как шлюз по умолчанию.
+    # Совпадает с сloud.prefix в ресурсе ClusterConfiguration
```
Comment (z9r5) [inline ee/candi/cloud-providers/openstack/docs/LAYOUTS_RU.md:336]:
```suggestion
    # Обязательный параметр, шлюз этой сети будет использован как шлюз по умолчанию.
    # Совпадает с сloud.prefix в ресурсе ClusterConfiguration.
```

**Review by z9r5 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [9181](https://github.com/deckhouse/deckhouse/pull/9181) to the branch [release-1.62](https://github.com/deckhouse/deckhouse/tree/release-1.62) successful!

**Pull request merged**